docs/spec/data-plane.md

```diff
@@ -0,0 +1,110 @@
+# Knative Eventing Data Plane Contracts
+
+## Introduction
+
+Developers using Knative Eventing need to know what is supported for delivery to
+user provided components that receive events. Knative Eventing defines contract
+for data plane components and we have listed them here.
+
+## Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in RFC2119.
+
+## Data plane contract for Sinks
+
+A **Sink** MUST be idempotent and MUST be able to handle duplicate events.
```

Comment:
I'd drop the MUST be idempotent here and just say that it must be able to handle duplicate events. Sink can choose to not be idempotent and deal with the deduping elsewhere.